src/Components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+
+type PropsType = {
+  setFilter: (filter: Filter) => void,
+  filter: Filter;
+  activeTodos: number;
+  completedTodos: number;
+  handleDeleteCompleted: () => void;
+};
+
+export const TodoFooter: React.FC<PropsType> = React.memo(
+  ({
+    setFilter,
+    filter,
+    activeTodos,
+    completedTodos,
+    handleDeleteCompleted,
+  }) => (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${activeTodos} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: filter === Filter.All },
+          )}
+          onClick={() => {
+            setFilter(Filter.All);
+          }}
```

Comment:
Do not use anonymous function. Create on `filterChangeHandler`. Apply in to all buttons

```
const filterChangeHandler = (filter: Filter) => () => {
  setFilter(filter)
}
```
and use 

`onClick={filterChangeHandler(Filter.All)}`